habitat/datasets/rearrange/samplers.py

```diff
@@ -0,0 +1,596 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+import random
+from abc import ABC, abstractmethod
+from typing import Dict, List, Optional, Tuple
+
+import magnum as mn
+
+import habitat.datasets.rearrange.sim_utilities as sutils
+import habitat_sim
+
+
+class SceneSampler(ABC):
+    @abstractmethod
+    def num_scenes(self):
+        pass
+
+    def reset(self) -> None:
+        pass
+
+    @abstractmethod
+    def sample(self):
+        pass
+
+
+class SingleSceneSampler(SceneSampler):
+    """
+    Returns a single provided scene using the sampler API
+    """
+
+    def __init__(self, scene: str) -> None:
+        self.scene = scene
+
+    def sample(self) -> str:
+        return self.scene
+
+    def num_scenes(self) -> int:
+        return 1
+
+
+class MultiSceneSampler(SceneSampler):
+    """
+    Uniform sampling from a set of scenes.
+    """
+
+    def __init__(self, scenes: List[str]) -> None:
```

Comment:
```suggestion
    def __init__(self, scenes: Sequence[str]) -> None:
```
This is actually the type you want.